src/core/shape_inference/include/strided_slice_shape_inference.hpp

```diff
@@ -23,26 +26,27 @@ void shape_infer(const StridedSlice* op,
     NODE_VALIDATION_CHECK(op, (input_shapes.size() == 3 || input_shapes.size() == 4) && output_shapes.size() == 1);
 
     const auto& input_shape = input_shapes[0];
-    const auto& begin_shape = input_shapes[1];
-    NODE_VALIDATION_CHECK(op,
-                          begin_shape.rank().compatible(1),
-                          "Begin input must be 1D (begin rank: ",
-                          begin_shape.rank(),
-                          ").");
 
-    const auto& end_shape = input_shapes[2];
-    NODE_VALIDATION_CHECK(op,
-                          end_shape.rank().compatible(1),
-                          "End input must be 1D (end rank: ",
-                          end_shape.rank(),
-                          ").");
+    const auto capitalize = [](std::string s) -> std::string {
+        if (!s.empty()) {
+            s.front() = std::toupper(s.front());
+        }
+        return std::move(s);
+    };
 
-    const auto& strides_shape = input_shapes.size() < 4 ? op->get_input_shape(3) : input_shapes[3];
-    NODE_VALIDATION_CHECK(op,
-                          strides_shape.rank().compatible(1),
-                          "Strides input must be 1D (strides rank: ",
-                          strides_shape.rank(),
-                          ").");
+    for (size_t i = 1; i < input_shapes.size(); ++i) {
+        const auto& shape_rank = input_shapes[i].rank();
+        NODE_VALIDATION_CHECK(op,
+                              shape_rank.compatible(1),
+                              capitalize(shape_names[i - 1]),
+                              " input must be 1D (",
+                              shape_names[i - 1],
```

Comment:
The `capitalize` lambda seems to be a little overkill, the names in the array can have already capitalized names.
Also it unnecessarily complicate the shape_infer code, as it's not related to logic, but the error message.
Overall in my opinion, there is nothing wrong with usage of input index instead of string name in this case.

Reply:
It was done to preserve original error message. But if is not must so I changed them a little to not use `capitalize` and remove this helper. Change names in array to be capital letter first.